docs/StakingPool.md

```diff
@@ -0,0 +1,289 @@
+
+```solidity
+struct Tranche {
+  uint128 stakeShares;
+  uint128 rewardsShares;
+}
+```
+
+| Parameter        | Description                           |
+|------------------|---------------------------------------|
+| `stakeShares`    | Tranche's share of the pool's stake   |
+| `rewardsShares`  | Tranche's share of the pool's rewards |
+
+```solidity
+uint public constant TRANCHE_DURATION = 91 days;
+uint public constant MAX_ACTIVE_TRANCHES = 8; // 7 whole quarters + 1 partial quarter
+```
+
+Formula for current tranche id:
+```solidity
+uint currentTrancheId = block.timestamp / TRANCHE_DURATION;
+```
+### Buckets
+Groupings used to manage rewards over time.
+
+```solidity
+uint public constant BUCKET_DURATION = 28 days;
+```
+
+```solidity
+uint public constant BUCKET_DURATION = 28 days;
+```
+Formula for current bucket id:
+```solidity
+    uint currentBucketId = block.timestamp / BUCKET_DURATION;
+```
+
+### Allocations
+Groupings of used capacity per tranche.
+
+## Mutative Functions
+
+### `depositTo`
+Allows users to deposit NXM into the pool, receiving stake and rewards shares in return. Supports deposits to specific tranches and allows adding to an existing deposit.
+
+```solidity
+function depositTo(uint amount, uint trancheId, uint requestTokenId, address destination) external;
+```
+| Parameter        | Description                                                                                                 |
+|------------------|-------------------------------------------------------------------------------------------------------------|
+| `amount`         | The amount to deposit.                                                                                      |
+| `trancheId`      | The ID of the tranche to deposit into.                                                                      |
+| `requestTokenId` | The ID of the request token (0 for a new deposit, or use an existing token ID to add to a previous deposit. |
+| `destination`    | The address to send the Staking NFT token to.                                                               |
+
+### `withdraw`
+Allows users to withdraw their stake and/or rewards from specific tranches, only from expired tranches.
+
+
+```solidity
+function withdraw(uint tokenId, bool withdrawStake, bool withdrawRewards, uint[] memory trancheIds) external;
+```
+
+| Parameter         | Description                                       |
+|-------------------|---------------------------------------------------|
+| `tokenId`         | The ID of the staking NFT token.                  |
+| `withdrawStake`   | Whether to withdraw the stake.                    |
+| `withdrawRewards` | Whether to withdraw the rewards.                  |
+| `trancheIds`      | The IDs of the expired tranches to withdraw from. |
+
+### `extendDeposit`
+Extends the duration of a deposit by moving it from an active tranche to a future tranche.
```

Comment:
"from an active tranche" -> false. tranche can be active or expired, doesn't matter